packages/deployments/contracts/contracts/core/connext/facets/NomadFacet.sol

```diff
@@ -184,183 +134,31 @@ contract NomadFacet is BaseConnextFacet {
     address[] memory routers = s.routedTransfers[transferId];
 
     // If fast transfer was made using portal liquidity, we need to repay
-    // FIXME: routers can repay any-amount out-of-band using the `repayAavePortal` method
+    // NOTE: routers can repay any-amount out-of-band using the `repayAavePortal` method
     // or by interacting with the aave contracts directly
     uint256 portalTransferAmount = s.portalDebt[transferId] + s.portalFeeDebt[transferId];
 
-    uint256 toDistribute = amount;
     uint256 pathLen = routers.length;
-    if (portalTransferAmount != 0) {
+    if (portalTransferAmount != 0 && pathLen != 1) {
       // ensure a router took on credit risk
-      if (pathLen != 1) revert NomadFacet__reconcile_noPortalRouter();
-      toDistribute = _reconcileProcessPortal(canonicalId, amount, token, transferId);
+      revert NomadFacet__reconcile_noPortalRouter();
     }
 
     if (pathLen != 0) {
       // fast liquidity path
       // Credit each router that provided liquidity their due 'share' of the asset.
-      uint256 routerAmt = toDistribute / pathLen;
-      for (uint256 i; i < pathLen; ) {
-        s.routerBalances[routers[i]][token] += routerAmt;
+      uint256 routerAmt = _amount / pathLen;
+      for (uint256 i; i < pathLen - 1; ) {
+        s.routerBalances[routers[i]][_localToken] += routerAmt;
         unchecked {
           ++i;
         }
       }
+      // The last router in the multipath will sweep the remaining balance to account for remainder dust.
+      uint256 toSweep = routerAmt + (_amount % pathLen);
+      s.routerBalances[routers[pathLen - 1]][_localToken] += toSweep;
     }
 
-    emit Reconciled(transferId, _origin, routers, token, amount, msg.sender);
```

Comment:
looks like we aren't emitting the `_origin` argument in this event anymore. any reason why? it's still being passed in (as the first arg) to `onReceive`.